### PR TITLE
(CDAP-16669) New Dataproc provisioner for existing cluster

### PR DESCRIPTION
**Pull request by chtyim**
This PR has two commits.

1. Refactor the existing Dataproc provisioner code to extract common implementation for the ephemeral and existing clusters.
   * In the process, also clean some clumsy code around private cluster logic
2. Added a new provisioner for existing Dataproc cluster

### CODE REVIEW

File: cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/ExistingDataprocProvisioner.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    String clusterName = properties.get("clusterName");
+    if (Strings.isNullOrEmpty(clusterName)) {
+      throw new IllegalArgumentException("Dataproc cluster name is missing");
+    }
+  }
+
+  @Override
+  protected String getClusterName(ProvisionerContext context) {
+    return context.getProperties().get("clusterName");
```
Comment (CuriousVini) [inline cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/ExistingDataprocProvisioner.java]: nit: can use `CLUSTER_NAME` instead

**Review by CuriousVini — APPROVED** [review]
lgtm 👍

File: cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/ExistingDataprocProvisioner.java (hunk 1 above)
Comment (chtyim) [reply cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/ExistingDataprocProvisioner.java]: fixed

**Comment by chtyim** [comment]
Build passed https://builds.cask.co/browse/CDAP-DUT7234

**Pull request merged**